### PR TITLE
Added RESTService tests

### PR DESCRIPTION
**Pull request by blazejbujala**

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -80,13 +84,13 @@ jacocoTestCoverageVerification {
     violationRules {
         rule {
             limit {
-                minimum = 0.7
+                minimum = 0.85
```
Comment (stanleyszysz) [inline app/build.gradle:87]: Doesn't it need to be separated into lines and branch level?
Maybe write to Radek to have a look at it?

Comment (blazejbujala) [reply app/build.gradle:87]: I think that if it is not seperated,  than it is being counted on both branch and lines together.

Comment (stanleyszysz) [reply app/build.gradle:87]: See the branch coverage in the jacoco report...
I had this problem :)

Comment (blazejbujala) [reply app/build.gradle:87]: Have checked jaccoco report and looks like only 'isInRepository(id)" method is cousing some losses in branch covarage. Thanks for the comment, will check this with Radek.

File: app/src/test/groovy/pl/futurecollars/invoicing/model/InvoiceTest.groovy [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    def "should be able to create instance of the class"() {
+
+        given:
+        def invoice = new Invoice(UUID.randomUUID(), LocalDate.now(), new Company(), new Company(), Arrays.asList())
+
+        when:
+        def expect = invoice
+
+        then:
+        expect != null
```
Comment (mtszpater) [inline app/src/test/groovy/pl/futurecollars/invoicing/model/InvoiceTest.groovy:18]: To be honest - I never create tests for basic java models. There is no logic to test. ;p I know that probably you did it to achieve high test coverage but in my opinion - it's not worth to do it in that way.

https://stackoverflow.com/questions/6197370/should-unit-tests-be-written-for-getter-and-setters

File: app/src/test/groovy/pl/futurecollars/invoicing/services/FileServiceTest.groovy [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+    def "should create file"() {
+        given:
+        def path = "app/src/test/resources/inv.json"
+
+        when:
+        fileService.readInvoicesFromFile(path)
+
+        then:
+        thrown(NullPointerException)
```
Comment (mtszpater) [inline app/src/test/groovy/pl/futurecollars/invoicing/services/FileServiceTest.groovy:17]: test 'should create file' so why it throws null pointer exception?

File: app/src/test/groovy/pl/futurecollars/invoicing/controllers/InvoiceControllerTest.groovy [hunk 4]
```diff
@@ -0,0 +1,254 @@
...
+                .andExpect(status().isOk())
+                .andReturn()
+                .getResponse()
+                .getContentAsString()
+
+        def id = jsonService.toObject(postResponse, Invoice.class).getId()
+
+        UUID updatedId
+        do updatedId = UUID.randomUUID()
+        while (updatedId == id)
```
Comment (mtszpater) [inline app/src/test/groovy/pl/futurecollars/invoicing/controllers/InvoiceControllerTest.groovy:242]: what do you want to do here?
```
      UUID updatedId
        do updatedId = UUID.randomUUID()
        while (updatedId == id)
```

I don't understand it - you are generating randomUUID until your updatedId is equals to ID retrievied from 
```
jsonService.toObject(postResponse, Invoice.class).getId()
```
? Why?

File: app/src/test/groovy/pl/futurecollars/invoicing/controllers/InvoiceControllerTest.groovy [hunk 5]
```diff
@@ -0,0 +1,254 @@
...
+                .andReturn()
+                .getResponse()
+                .getContentAsString()
+
+        def id = jsonService.toObject(postResponse, Invoice.class).getId()
+        invoice1.setId(id)
+
+        UUID updatedId
+        do updatedId = UUID.randomUUID()
+        while (updatedId == id)
```
Comment (mtszpater) [inline app/src/test/groovy/pl/futurecollars/invoicing/controllers/InvoiceControllerTest.groovy:177]: what do you want to do here?
```
      UUID updatedId
        do updatedId = UUID.randomUUID()
        while (updatedId == id)
```

I don't understand it - you are generating randomUUID until your updatedId is equals to ID retrievied from 
```
jsonService.toObject(postResponse, Invoice.class).getId()
```
? Why?

File: app/src/test/groovy/pl/futurecollars/invoicing/config/InMemoryConfigurationTest.groovy [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+    }
+
+    def "should init jsonService"() {
+        expect:
+        inMemoryConfiguration.jsonService()
+    }
+
+    def "should init jsonServiceList"() {
+        expect:
+        inMemoryConfiguration.jsonServiceList()
```
Comment (mtszpater) [inline app/src/test/groovy/pl/futurecollars/invoicing/config/InMemoryConfigurationTest.groovy:28]: ```InMemoryConfigurationTest ``` unnecessary test class. It tests nothing. Please remove it ;)

File: app/src/test/groovy/pl/futurecollars/invoicing/controllers/InvoiceControllerTest.groovy (hunk 4 above)
Comment (blazejbujala) [reply app/src/test/groovy/pl/futurecollars/invoicing/controllers/InvoiceControllerTest.groovy:242]: To be sure that I have updateId different form id of invocie in database, this test is checking what happen if we try to update not exeisting invoice.

File: app/src/test/groovy/pl/futurecollars/invoicing/services/FileServiceTest.groovy (hunk 3 above)
Comment (blazejbujala) [reply app/src/test/groovy/pl/futurecollars/invoicing/services/FileServiceTest.groovy:17]: I have changed name of test method

File: app/src/test/groovy/pl/futurecollars/invoicing/model/InvoiceTest.groovy (hunk 2 above)
Comment (blazejbujala) [reply app/src/test/groovy/pl/futurecollars/invoicing/model/InvoiceTest.groovy:18]: have delated InvocieTest

File: app/src/test/groovy/pl/futurecollars/invoicing/controllers/InvoiceControllerTest.groovy (hunk 5 above)
Comment (blazejbujala) [reply app/src/test/groovy/pl/futurecollars/invoicing/controllers/InvoiceControllerTest.groovy:177]: It's generating randomUUID while updatedId is equal to id of invocie that is in database, to be sure that updateId (id of invoice to be updated) is different form id of invocie in database. This test is checking what happen if we try to update not exeisting in database invoice, so we need to be sure that updatedId is diffrent from id of invoice in database. I think it's ok, am I wrong ?

File: app/src/test/groovy/pl/futurecollars/invoicing/config/InMemoryConfigurationTest.groovy (hunk 6 above)
Comment (blazejbujala) [reply app/src/test/groovy/pl/futurecollars/invoicing/config/InMemoryConfigurationTest.groovy:28]: Delated :)

File: app/src/test/groovy/pl/futurecollars/invoicing/services/FileServiceTest.groovy (hunk 3 above)
Comment (mtszpater) [reply app/src/test/groovy/pl/futurecollars/invoicing/services/FileServiceTest.groovy:17]: but why it should throw an exception? please add more details to test e.g:

"should throw exception if database file doesn't exist"

File: app/src/test/groovy/pl/futurecollars/invoicing/model/InvoiceTest.groovy (hunk 2 above)
Comment (mtszpater) [reply app/src/test/groovy/pl/futurecollars/invoicing/model/InvoiceTest.groovy:18]: i still see it in the commit:D

Comment (blazejbujala) [reply app/src/test/groovy/pl/futurecollars/invoicing/model/InvoiceTest.groovy]: corrected with third commit (have deleted wrong test class)

File: app/src/test/groovy/pl/futurecollars/invoicing/services/FileServiceTest.groovy (hunk 3 above)
Comment (blazejbujala) [reply app/src/test/groovy/pl/futurecollars/invoicing/services/FileServiceTest.groovy:17]: have changed name of test to "should read not existing file and throw exception"

File: app/src/test/groovy/pl/futurecollars/invoicing/controllers/InvoiceControllerTest.groovy (hunk 4 above)
Comment (blazejbujala) [reply app/src/test/groovy/pl/futurecollars/invoicing/controllers/InvoiceControllerTest.groovy]: after our conversation, I have deleted this part, now test is shorter and more transparent

File: app/src/test/groovy/pl/futurecollars/invoicing/controllers/InvoiceControllerTest.groovy (hunk 5 above)
Comment (blazejbujala) [reply app/src/test/groovy/pl/futurecollars/invoicing/controllers/InvoiceControllerTest.groovy:177]: after our conversation, I have deleted this part, now test is shorter and more transparent

**Review by mtszpater — APPROVED** [review]

**Review by stanleyszysz — APPROVED** [review]

**Pull request merged**